src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -306,6 +331,9 @@ await ids.Select(id => _releaseFileRepository.CheckFileExists(releaseId, id, Fil
                                                         archiveFile.DataFileName))
                                                 .OnSuccess(async () =>
                                                 {
+                                                    var subjectId = await CreateReleaseAndSubjectHierarchy(release,
```

Comment:
I think you should refactor `CreateReleaseAndSubjectHierarchy` and even the hydrating of the Release above to a new service or repository for creating the Subject as it's over complicated this service which should only concerned with data files.